### ISSUE TITLE
Table name lookup is broken when two structs have the same name

### ISSUE THREAD

**Issue by aeneasr** [gobuffalo/pop#477]
## Description

Caches are used across the projects, for example, to save time when looking up a model's table name. That however causes issues if two structs are using the same name.

### Steps to Reproduce the Problem

Create two models with the same struct name in two packages and associations. Using eager `find` will confuse the SQL Builder and generate nonsense.

### Expected Behavior

Caches should be able to deal with structs that have the same name.

### Actual Behavior

Unable to query because the cache is non-deterministic. Potential culprits:

https://github.com/gobuffalo/pop/blob/master/model.go#L16

https://github.com/gobuffalo/pop/blob/master/model.go#L77-L86

https://github.com/gobuffalo/pop/blob/master/model.go#L89-L115

### Info

Latest master

**Comment by aeneasr** [comment]
Seems like that's the problem. I have two models: `login.RequestMethod` and `registration.RequestMethod` with individual tables and columns. The cache only has `RequestMethod` when `TableName` is called: `cache: map[RequestMethod:selfservice_registration_request_methods]`.

By making cache keys unique (adding pkg path) this can be resolved with a few lines of code (will pull up PR):

```
map[github.com/ory/kratos/selfservice/flow/login.RequestMethod:selfservice_login_request_methods github.com/ory/kratos/selfservice/flow/registration.RequestMethod:selfservice_registration_request_methods]
```

**Comment by aeneasr** [comment]
Resolved by #479
